CONTRIBUTING.md

```diff
@@ -1,49 +1,151 @@
+
+In this example, we will stick to "espace de nom" in the rest of the page:
+```diff
+- Components like these are known as namespaced components. Namespacing allows us to organize our components by folders according to their purpose.
++ Les composants comme ceux-ci sont appelés composants à espace de nom (_namespaced components_). Les espaces de nom permettent d'organiser les composants par dossiers selon leur fonction.
 ```
-git remote add upstream https://github.com/dazzlingfugu/ember-fr-guides-source.git
-git checkout master
-git pull upstream master
+
+In this example, we will go for "store" and "record" in the rest of the page:
+```diff
+- The EmberData store provides an interface for retrieving records of a single type. Use `store.findRecord()` to retrieve a record.
++ Le _store_ (magasin, réserve) d'EmberData fournit une interface pour retrouver les _records_ (enregistrements) d'un seul type. Utilisez `store.findRecord()` pour récupérer un _record_.
+```
+
+If the literal translation of a term is _reeeeally_ convoluted, don't hesitate to be wordy about it. The purpose is to make this understandable to non-English speakers:
+
+```diff
+- This method is also known as the _model hook_.
++ Cette méthode est aussi appelée le _model hook_ (_hook_ signifie littéralement "un crochet" mais se traduit mal dans un contexte technique. On peut voir le _hook_ comme une fonction mise à disposition et exécutée en interne par le framework, et dont on définit le contenu, on y "accroche" du code).
+```
+
+### 3. Don't translate articles titles / translate descriptions
+
+Some pages of the Guides reference articles, introduced with a title and sometimes a description. If this resource is also available in French (for instance, a lot of pages from MDN website are translated), then use the French URL in the translation (be careful with that MDN example though, because some pages are not translated entirely and it can be better to keep the English).
+
+If the resource is not translated or if the translation is incomplete, keep the English link and translate only the description, not the title of the resource itself, because we need to make clear to the reader that it's an English document:
+```diff
+- [Using ARIA:](https://www.w3.org/TR/using-aria/) a practical guide for developers on how to add accessibility information to HTML element
++ <!-- spell ignore -->
++ [Using ARIA](https://www.w3.org/TR/using-aria/) : un guide pratique pour ajouter aux éléments HTML des informations accessibles.
+```
+
+### 4. Set non-breakable spaces
+
+`&nbsp;` is the HTML sign for "non-breakable space" and can be used in markdown as well. If the space is not set non-breakable, then a punctuation sign like ":" or ";" could be separated from the previous word and moved to the next line, which is incorrect.
+```diff
+- Using ARIA: a practical guide (...)
++ Using ARIA&nbsp;: un guide pratique (...)
+```
```

Comment:
I am still **really doubtful** about this whole section (I know I already commented about that, somewhere).
Besides you use the conditional "could be separated" so it is clear "this can sometimes happen, sometimes not".

I have never seen such an overuse of `&nbsp;` in Markdown files, even when their content is in French.

Markdown is not that weird, if there is one single space it is and will be respected.
However, 2 spaces at the end of a line _can_ create a line-break, that depends on the tool rendering the Markdown file. This is known as hard-break and soft-breaks:
- https://spec.commonmark.org/0.30/#hard-line-breaks
- https://spec.commonmark.org/0.30/#soft-line-breaks

We, as reviewers, can detect these by enabling the "Show whitespace changes". But I am confident we should not encounter many situations where people put 2 spaces or not between a word and a ponctuation sign.

One quick example:
https://raw.githubusercontent.com/vuejs-translations/docs-fr/8851ab5351b4015d8c3b5404fa17d3e3da23a77b/src/guide/scaling-up/sfc.md

Reply:
I have always marked my non-breakable spaces, no matter the content format 🤷‍♀️ 

I don't see how the line-breaking samples are related to that, to be honest. To me, the main reason why people don't write non-breakable spaces is because they are lazy or ignorant about the fact punctuation marks can't be moved to the next line in French.

After our first discussion about this, I tested the app without `&nbsp;` and I am totally able to make the render incorrect on some screen sizes, so I don't see why we shouldn't put them, it's not a big deal, it can even be done with a search & replace.